src/Queue/Task/EmailTask.php

```diff
@@ -109,6 +109,31 @@ public function run(array $data, int $jobId): void {
 			throw new QueueException('Queue Email task called without settings data.');
 		}
 
+		if ($data['settings'] === 'mailer') {
+			if (!isset($data['mailer'])) {
+				throw new QueueException('Queue Email task called without mailer data.');
+			}
+			Configure::write('Queue.mailerClass', $data['mailer']);
+			$this->mailer = $this->getMailer();
+
+			try {
+				$this->mailer->setTransport($data['transport'] ?? 'default');
+				$result = $this->mailer->send($data['action'], $data['vars'] ?? []);
+			} catch (\Throwable $e) {
+				$error = $e->getMessage();
+				$error .= ' (line ' . $e->getLine() . ' in ' . $e->getFile() . ')' . PHP_EOL . $e->getTraceAsString();
+				Log::write('error', $error);
+
+				throw $e;
+			}
+
+			if (!$result) {
+				throw new QueueException('Could not send email.');
+			}
+
+			return;
+		}
+
 		/** @var \Cake\Mailer\Message|null $message */
 		$message = $data['settings'];
```

Comment:
$message can already be \Cake\Mailer\Message|null
It would be cleaner to add Mailer as object type in there as well, not relying on the string to be mailer

Reply:
so basically I just left the EmailTask as it is to leave it as the legacy mail task.
Any new (non serialized object) mail logic should happen in the MailerTask.
But I would generally stay away from serialization since one should never pass unsafe data to unserialize.